### PR TITLE
Example using Vault to fetch wnd password

### PR DESCRIPTION
**Pull request by IljaN**
owncloud/windows_network_drive#321

### CODE REVIEW

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 1]
```diff
@@ -584,9 +584,21 @@ Third party password managers can also be integrated. The only requirement is th
...
+
+* See https://learn.hashicorp.com/collections/vault/getting-started
+on how to setup Vault as your secrets store. Then use something like the following
+
+[source,console,subs="attributes+"]
+----
+vault kv get -field=password secret/samba | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
+----
+
+Use Vault's ACLs to limit access of the token. Destroy the token after starting the service
```
Comment (C0rby) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:597]: > Destroy the token after starting the service during boot with systemd.

What do you mean by that?

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 2]
```diff
@@ -584,9 +584,21 @@ Third party password managers can also be integrated. The only requirement is th
...
 
-As an example:
+Example using "Vault" as secrets store
+
+* See https://learn.hashicorp.com/collections/vault/getting-started
+on how to setup Vault as your secrets store. Then use something like the following
+
+[source,console,subs="attributes+"]
+----
+vault kv get -field=password secret/samba | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
```
Comment (C0rby) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:594]: Can we change the flags of the wnd app? `--password-file` looks wrong.

Comment (C0rby) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:594]: Of course this needs to be done in the app not the docs. :)

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 3]
```diff
@@ -584,9 +584,21 @@ Third party password managers can also be integrated. The only requirement is th
 password in plain text somehow. If not, additional operations might be required to get the password as
 plain text and inject it in the listener.
 
-As an example:
+Example using "Vault" as secrets store
+
+* See https://learn.hashicorp.com/collections/vault/getting-started
+on how to setup Vault as your secrets store. Then use something like the following
+
+[source,console,subs="attributes+"]
```
Comment (C0rby) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:592]: What does that mean?

**Comment by mmattel** [comment]
I have seen a possible improvement, but I am in the road right now. Will do it later.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 4]
```diff
@@ -584,9 +584,21 @@ Third party password managers can also be integrated. The only requirement is th
 password in plain text somehow. If not, additional operations might be required to get the password as
 plain text and inject it in the listener.
 
-As an example:
+Example using "Vault" as secrets store
+
+* See https://learn.hashicorp.com/collections/vault/getting-started
```
Comment (mmattel) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:589]: can you make this link a macro as we do with the other links?
see on top of the page
like: `:vault-getting-started-url: https://learn.hashicorp.com/collections/vault/getting-started`
and here: `See the {vault-getting-started-url}[get started here guide]

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 1 above)
Comment (IljaN) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:597]: Vault requires an access token, so theoretically one could provide it at boot-time to fetch the secret and then remove the token after the secret was fetched. The secret would reside in memory of the running wnd process only.  Not sure about this one either? Maybe too convoluted?

Comment (C0rby) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:597]: But which token do you mean?
The wnd secret? That should stay in the vault, right?
Or the access token to the vault? But how would it be provided? Manually by a person or is this an automated step?
I probably know too little about the vault setup process... :see_no_evil:

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 3 above)
Comment (IljaN) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:592]: This is adoc markup

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 1 above)
Comment (IljaN) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:597]: Ah right, I rewword

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 3 above)
Comment (C0rby) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:592]: I now I get it.. :sweat_smile:

**Comment by phil-davis** [comment]
@IljaN what is happejing with this docs PR?

I noticed because it is in my review requests.

CC @mmattel

**Comment by phil-davis** [comment]
I took myself off the review requests. Feel free to put me back on when someone is ready to merge this.

**Comment by mmattel** [comment]
@IljaN @C0rby (and @EParzefall @phil-davis)
I have reworked that section and would please you to give some feedback on the changes made.

For the already existing sections of plainpass and base64, I have not tried them myself if this is working or not, but after reading I expect this to be an idea-sampler...

For a good start, go into the viewing mode (View file) to see how the section will render (without resolving internal variables of course)
image

**Comment by IljaN** [comment]
LGTM :+1:

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 5]
```diff
@@ -659,43 +663,56 @@ NOTE: The password will be reset on the next request, regardless of the flag set
...
+==== HashiCorp Vault
+
+This example uses {vaultproject-url}[Vault] as secrets store. See {hashicorp-url}[HCP Vault] on how to setup the secrets store. Then use something like the following:
+
+[source,console,subs="attributes+"]
+----
+vault kv get -field=password secret/samba | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
+----
+
+Use Vault's ACLs to limit access of the token. Destroy the token after starting the service during boot with systemd.
```
Comment (phil-davis) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:707]:
```suggestion
Use Vault's ACLs to limit access to the token. Destroy the token after starting the service during boot with systemd.
```

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 6]
```diff
@@ -659,43 +663,56 @@ NOTE: The password will be reset on the next request, regardless of the flag set
...
+* You can go through {pass-url}[manage passwords from the command line] to setup the keyring for whoever will fetch the password (probably root) and then use something like the following
 
 [source,console,subs="attributes+"]
 ----
 pass the-password-name | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
 ----
 
+==== HashiCorp Vault
+
+This example uses {vaultproject-url}[Vault] as secrets store. See {hashicorp-url}[HCP Vault] on how to setup the secrets store. Then use something like the following:
```
Comment (phil-davis) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:700]:
```suggestion
This example uses {vaultproject-url}[Vault] as the secrets store. See {hashicorp-url}[HCP Vault] on how to setup the secrets store. Then use something like the following:
```

**Review by phil-davis — COMMENTED** [review]
Looks good - just 2 tiny English suggestions.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 7]
```diff
@@ -659,43 +663,56 @@ NOTE: The password will be reset on the next request, regardless of the flag set
...
 ----
 cat /tmp/plainpass | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
 ----
 
-This provides a bit more security because the `/tmp/plainpass` password should be owned by root and only
-root should be able to read the file (0400 permissions); Apache, particularly, shouldn't be able to read it.
-It's expected that root will be the one to run this command.
+==== base64
+
+Similar to plainpass, the contents in this case gets encoded in the {base64-url}[Base64 format]. There's not much security, but it has additional obfuscation.
```
Comment (EParzefall) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:679]:
```suggestion
Similar to plainpass, the content in this case gets encoded in the {base64-url}[Base64 format]. There's not much security, but it has additional obfuscation.
```

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 8]
```diff
@@ -659,43 +663,56 @@ NOTE: The password will be reset on the next request, regardless of the flag set
...
-password in plain text somehow. If not, additional operations might be required to get the password as
-plain text and inject it in the listener.
+Example using "pass"
 
-As an example:
-
-* You can use "pass" as a password manager.
-* You can go through http://xmodulo.com/manage-passwords-command-line-linux.html
-to setup the keyring for whoever will fetch the password (probably root) and then use something like the following
+* You can go through {pass-url}[manage passwords from the command line] to setup the keyring for whoever will fetch the password (probably root) and then use something like the following
```
Comment (EParzefall) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc:691]:
```suggestion
* You can go through {pass-url}[manage passwords from the command line] to set up the keyring for whoever will fetch the password (probably root) and then use something like the following:
```

**Review by EParzefall — APPROVED** [review]
LGTM except for the typos

**Review by phil-davis — APPROVED** [review]
LGTM

**Pull request merged**